### PR TITLE
Add legend touch event for touchable device

### PR DESCRIPTION
**Pull request by wuliupo**
Add legend touch event for touchable device, like mobile, pad.
The touchstart event replaces the click event.
The taphold (more than 1 second touch) event replaces the dblclick event.

### CODE REVIEW

File: examples/pieChart.html [hunk 1]
```diff
@@ -2,6 +2,8 @@
 <html>
 <head>
     <meta charset="utf-8">
+    <title>nvd3 example - pie chart</title>
+    <meta content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0" name="viewport" />
```
Comment (wuliupo) [inline examples/pieChart.html]: add viewport meta for touch device

File: src/models/legend.js [hunk 2]
```diff
@@ -187,6 +182,44 @@ nv.models.legend = function() {
 
         return chart;
     }
+    
+    function toggleItem(d,i,data){
```
Comment (wuliupo) [inline src/models/legend.js]: the two function toggleItem and checkItemOnly are copied from old event for re-usage, between common click/dblclick and touchstart/taphold.

File: src/models/legend.js [hunk 3]
```diff
@@ -38,43 +38,38 @@ nv.models.legend = function() {
...
-                        }
-                        else {
-                            d.disabled = !d.disabled;
-                            if (data.every(function(series) { return series.disabled})) {
-                                //the default behavior of NVD3 legends is, if every single series
-                                // is disabled, turn all series' back on.
-                                data.forEach(function(series) { series.disabled = false});
-                            }
+                });
+            if(isTouchable) {
```
Comment (wuliupo) [inline src/models/legend.js]: if PC device, use click/dblclick.
if touchable device, use touchstart/taphold.
Taphold is a long touch more than 1 second.

File: src/models/legend.js [hunk 4]
```diff
@@ -38,43 +38,38 @@ nv.models.legend = function() {
...
+                seriesEnter.on("touchstart", function(d,i){
+                    seriesEnter.attr("tap_triggered", "0");
```
Comment (wuliupo) [inline src/models/legend.js]: there data function cannot be used due nvd3 listener.

**Comment by liquidpele** [comment]
Cool, will review this when I get a chance since it's not a trivial change.

**Comment by wuliupo** [comment]
Hi @liquidpele, 

You can ignore two functions added, which are the old click, dblclick event,
```
function toggleItem(d,i,data)   // line 186
function checkItemOnly(d,i,data)   // line 209
```

Please view the touch event,
```
if(isTouchable) {   // line 42
//...
}
```

**Comment by liquidpele** [comment]
Hey there, sorry but there was a lot of movement and this looks to be unmergable at this point.  

Can you submit your changes to the novus repo instead of this one?

**Pull request closed without merging**